base/iddict.jl

```diff
@@ -83,18 +84,31 @@ function sizehint!(d::IdDict, newsz)
     rehash!(d, newsz)
 end
 
-function setindex!(d::IdDict{K,V}, @nospecialize(val), @nospecialize(key)) where {K, V}
+function ht_keyindex!(d::IdDict{K,V}, @nospecialize(key)) where {K, V}
```

Comment:
This does not appear to follow the API for `ht_keyindex` particularly well. It is closer to `ht_keyindex2!`, though with minor deviations in the return that it looks like could be changed here to match?